Composer/packages/server/src/server.ts

```diff
@@ -26,92 +26,92 @@ import pluginLoader from './services/pluginLoader';
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const session = require('express-session');
 
-const app: Express = express();
-app.set('view engine', 'ejs');
-app.set('view options', { delimiter: '?' });
-app.use(compression());
-
-app.use(bodyParser.json({ limit: '50mb' }));
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(session({ secret: 'bot-framework-composer' }));
-app.use(pluginLoader.passport.initialize());
-app.use(pluginLoader.passport.session());
-
-// make sure plugin has access to our express...
-pluginLoader.useExpress(app);
-
-// load all the plugins that exist in the folder
-pluginLoader.loadPluginsFromFolder(path.resolve(__dirname, '../../../plugins')).then(() => {
-  const { login, authorize } = getAuthProvider();
-
-  const CS_POLICIES = [
-    "default-src 'none';",
-    "font-src 'self' https:;",
-    "img-src 'self' data:;",
-    "base-uri 'none';",
-    "connect-src 'self';",
-    "frame-src 'self' bfemulator: https://login.microsoftonline.com https://*.botframework.com;",
-    "worker-src 'self';",
-    "form-action 'none';",
-    "frame-ancestors 'self';",
-    "manifest-src 'self';",
-    'upgrade-insecure-requests;',
-  ];
-
-  app.all('*', function(req: Request, res: Response, next: NextFunction) {
-    res.header('Access-Control-Allow-Origin', '*');
-    res.header('Access-Control-Allow-Methods', 'GET,HEAD,OPTIONS,POST,PUT,DELETE');
-    res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization');
-
-    if (process.env.ENABLE_CSP === 'true') {
-      req.__nonce__ = crypto.randomBytes(16).toString('base64');
-      res.header(
-        'Content-Security-Policy',
-        CS_POLICIES.concat([
-          `script-src 'self' 'nonce-${req.__nonce__}';`,
-          // TODO: use nonce strategy after addressing issues with monaco-editor pacakge
-          "style-src 'self' 'unsafe-inline'",
-          // `style-src 'self' 'nonce-${req.__nonce__}';`,
-        ]).join(' ')
-      );
-    }
+export async function start(pluginDir?: string) {
+  const clientDirectory = path.resolve(require.resolve('@bfc/client'), '..');
```

Comment:
Can you explain why this is necessary?

Reply:
I initially did this to back out of what Chris Anderson did in his branch, and also get rid of the need for the `/client/` build script to copy the build files over to the `/server/` package.

It seemed more straightforward to just take a dependency on `/client/` from `/server/`, because technically it does have a hard dependency on `/client/`, even though it's implicit (client must copy its built files over for server to work properly).

Also, in the Composer Electron meeting that started the hand-off from Chris to me, we discussed a potential future where we would have `composer-client`, `composer-server` (core that mounts the routes), and `composer-electron` (or whatever your wrapper would be -- Restify, Express, etc.) packages that would be pieced together to start Composer. In this model, copying files over from client to server doesn't really fit.

I'm ok with reverting this change and re-enabling the file copying if that's what we want. I was just trying to think ahead and set this up for a model where these packages aren't always local workspaces.

Reply:
I'm not too sure about this approach. I agree that there is an implicit dependency on the client, but really anything can inject static assets to the public directory and have express serve them. The reason I'm questioning this is the changes to the dockerfile. It pretty clearly shows the additional complexity that is required on the host machine (by needing to copy client, extension, ui-plugins, etc). I don't want to block this PR on that so we can discuss more later.

Reply:
Yeah, the additions to the docker file made me a little hesitant as well. Happy to discuss it further once this is in.